### PR TITLE
Analytics api basics and setup

### PR DESCRIPTION
**Pull request by bapierre**
Created the basic model and api to access a Player's data
Added tests
Added Integration test
Added application configuration for lightweight database

### CODE REVIEW

File: analytics/src/main/java/brocoeur/example/analytics/model/Player.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package brocoeur.example.analytics.model;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Player {
```
[the remark this answers is not in the document]
Comment (bapierre) [reply analytics/src/main/java/brocoeur/example/analytics/model/Player.java:9]: need to be public

File: analytics/src/main/java/brocoeur/example/analytics/controller/PlayerController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @Autowired
+    PlayerRepository playerRepository;
+
+    @GetMapping("/users")
+    ResponseEntity<Player> getAllPlayers(){
+        return new ResponseEntity(playerRepository.findAll(), HttpStatus.OK);
+    }
+
+    @GetMapping("/Players/{id}")
```
[the remark this answers is not in the document]
Comment (bapierre) [reply analytics/src/main/java/brocoeur/example/analytics/controller/PlayerController.java:23]: fixed

File: analytics/src/main/java/brocoeur/example/analytics/controller/PlayerController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        return new ResponseEntity(playerRepository.findAllById(ids),HttpStatus.OK);
+    }
+
+    @PostMapping("/Players")
+    ResponseEntity createPlayer(@RequestBody Player player){
+        return new ResponseEntity(playerRepository.save(player),HttpStatus.CREATED);
+    }
+
+    @DeleteMapping("/Players/{id}")
+    ResponseEntity updatePlayer(@PathVariable(value = "id") Long id){
```
[the remark this answers is not in the document]
Comment (bapierre) [reply analytics/src/main/java/brocoeur/example/analytics/controller/PlayerController.java:39]: fixed

File: analytics/src/test/java/brocoeur/example/analytics/controller/PlayerControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+public class PlayerControllerIntegrationTest {
+
+    @Autowired
+    private PlayerController playerController;
+
+    @Test
+    void contextLoadRepositoryTest(){
+        //given
+        //when
+        //then
```
[the remark this answers is not in the document]
Comment (bapierre) [reply analytics/src/test/java/brocoeur/example/analytics/controller/PlayerControllerIntegrationTest.java:21]: I keep it that way

File: analytics/src/test/java/brocoeur/example/analytics/controller/PlayerControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Collections;
+
+@SpringBootTest
+public class PlayerControllerTest {
```
[the remark this answers is not in the document]
Comment (bapierre) [reply analytics/src/test/java/brocoeur/example/analytics/controller/PlayerControllerTest.java:13]: fixed

**Review by gaugerard — APPROVED** [review]

**Comment by gaugerard** [comment]
Same logic implemented in analytics already. This code is out of date.

**Pull request closed without merging**